### ISSUE TITLE
reconstruction difference with more padding

### ISSUE THREAD

**Issue by McHaillet** [teamtomo/torch-fourier-slice#24]
Putting this here for reference about the effect of padding on reconstruction quality. The example is a small subtomogram from the SHREC dataset with initial box size of 64. Each image is mean projected after reconstruction to improve the contrast.

First a reconstruction with the current default padding (v0.2.0), `p = images.shape[-1] // 4`, box size is 96:

<p align="center">
  <img src="https://github.com/user-attachments/assets/ab718f37-f5e0-4621-915d-632245305c3f" width="300">
</p>

This is with exaggerated padding `p = images.shape[-1] * 2`, box size is 320:

<p align="center">
  <img src="https://github.com/user-attachments/assets/956b65f9-3924-4a58-8481-75db2d2dd868" width="300">
</p>

This is with padding `p = images.shape[-1]`, box size is 192;

<p align="center">
  <img src="https://github.com/user-attachments/assets/74d77792-7e01-4159-ae0d-bb2aeece477a" width="300">
</p>

More padding definitely seems better, but it comes at the expense of computation. Maybe a default of `p = images.shape[-1]` might be better though!

**Comment by jdickerson95** [comment]
I played around with this before and didn't see much difference in 2dtm scores, but I could also low pass filter to 3 Angstroms (1 A/px) without compromising the scores.
Have you done FRC's between these to see at what fraction of Nyquist you need a particular amount of padding?

**Comment by alisterburt** [comment]
Weird! Do any of the larger pad factors produce non-even dimensions? This might not be properly accounted for if so, but a quick glance at the implementation implies it should be okay...

**Comment by alisterburt** [comment]
idea: placing data in a padded volume reduces the support of each point as a fraction of the entire volume... when number of projections is small like this then I guess the fraction of total volume which has data decreases as the padding increases - this would explain the drop in FSC

validation experiment would be to check with a much larger number of projections, there FSC should get better with increased padding 🙂

**Comment by dtegunov** [comment]
Sorry, I'm still lost. Regardless, if the back-projection operates in Fourier space (I assume; haven't checked code), the impact of padding should be equal across the entire frequency range. The difference will be between the center and the periphery in real space.

**Comment by McHaillet** [comment]
Yes, both the forward and back projection are done in Fourier space.

For full clarity, on [this branch](https://github.com/McHaillet/torch-fourier-slice/tree/increase_padding), I ran this code to generate the FSC curves:

```
import matplotlib.pyplot as plt
import mrcfile
import torch
from scipy.stats import special_ortho_group
from torch_fourier_shell_correlation import fsc

from torch_fourier_slice import backproject_2d_to_3d, project_3d_to_2d

N_IMAGES = 50
torch.manual_seed(42)

# load a volume and normalise
volume = torch.tensor(
    mrcfile.read("/home/marten/data/datasets/emdb/test/emd_48372_10A.mrc")
)
volume -= torch.mean(volume)
volume /= torch.std(volume)

# random rotation matrices for projection (operate on xyz column vectors)
rotations = torch.tensor(
    special_ortho_group.rvs(dim=3, size=N_IMAGES, random_state=42)
).float()

fig, ax = plt.subplots()

for x in [1.0, 1.5, 2.0, 3.0, 4.0]:
    # make projections
    projections = project_3d_to_2d(
        volume,
        rotation_matrices=rotations,
        pad_factor=x,
    )  # (b, h, w)

    # reconstruct volume from projections
    reconstruction = backproject_2d_to_3d(
        images=projections,
        rotation_matrices=rotations,
        pad_factor=x,
    )
    reconstruction -= torch.mean(reconstruction)
    reconstruction = reconstruction / torch.std(reconstruction)

    _fsc = fsc(volume, reconstruction)[1:]
    ax.plot(_fsc, label=f"{x}")

ax.legend()
plt.show()
```

**Comment by McHaillet** [comment]
> validation experiment would be to check with a much larger number of projections, there FSC should get better with increased padding 🙂

Good one, if I run with 1000 random projections instead of 100 it does behave:

![Image](https://github.com/user-attachments/assets/41988368-0589-4e7f-81a6-5caca40b5f5c)

**Comment by McHaillet** [comment]
> idea: placing data in a padded volume reduces the support of each point as a fraction of the entire volume... when number of projections is small like this then I guess the fraction of total volume which has data decreases as the padding increases - this would explain the drop in FSC

I am fully getting what you mean but it seems correct 😅 I get your intuition but should this not be cancelled out when we unpad after doing the reconstruction?

**Comment by McHaillet** [comment]
FYI I merged the PR and set the pad_factor to 2.0 as a default. Leaving the issue open to continue the discussion.
